### ISSUE TITLE
Failing threading.SleepEx.test2

### ISSUE THREAD

**Issue by janvorli** [dotnet/coreclr#8321]
The test has failed in the CI test run below with the following error:
`Expected thread to sleep for 2000 ms (and not be interrupted). Average delta: 858993393 ms, acceptable delta: 150`
https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_centos7.1_prtest/436/
It seems we have a bug in the delta computation. Based on the test start times in the log, there was no delay in test execution.

**Comment by janvorli** [comment]
@kount, can you take a look please?

**Comment by AndyAyersMS** [comment]
Maybe unrelated but https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_centos7.1_prtest/3351 failed on
```
threading.Sleep.test1 (from (empty))
The sleep function slept for 2353 ms when it should have slept for 2000 ms
```

**Comment by danmosemsft** [comment]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_centos7.1_prtest/3516/testReport/junit/threading/SleepEx/test2/

**Comment by danmosemsft** [comment]
@kouvel maybe write out the timestamp before and after each sleep, at least in the failure case, so we can see whehter we're getting reasonable numbers?

**Comment by AndyAyersMS** [comment]
Another instance: https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_centos7.1_prtest/3608/

**Comment by adityamandaleeka** [comment]
I just ran into a similar issue, but in this case, the time it slept for was slightly less than expected:
```
The sleep function slept for 1984 ms when it should have slept for 2000 ms
```
https://ci.dot.net/job/dotnet_coreclr/job/master/job/debug_centos7.1_prtest/3695/testReport/junit/threading/SleepEx/test1/

**Comment by pgavlin** [comment]
(continuing discussion from #10261)

> This is a general problem with all of the wait tests that verify that a wait completes in reasonable time, especially when the machine is overloaded running other tests in parallel. These tests are useful when bringing up new systems and a sufficient buffer should be included when expecting wait completion. The buffer needs to be adjusted, for that part I'll consider this a duplicate of #8321.

The problem here is that the test is fundamentally unable to guarantee that it will pass no matter the buffer time, not that the buffer time is too small. The only property that these sorts of tests can check while guaranteeing that they pass is the property that a wait for `x` units of time does not return before `x` units of time have elapsed. At the very least, these tests should not attempt to verify that a wait completes "in a reasonable amount of time" when running in CI (since CI machines may be heavily loaded).

**Comment by mikedn** [comment]
Another one, this time on OS X: https://ci.dot.net/job/dotnet_coreclr/job/master/job/checked_osx10.12_prtest/162/

**Comment by kouvel** [comment]
> The only property that these sorts of tests can check while guaranteeing that they pass is the property that a wait for x units of time does not return before x units of time have elapsed

It's documented on Windows that waits can return before the specified duration as well.

> The problem here is that the test is fundamentally unable to guarantee that it will pass no matter the buffer time, not that the buffer time is too small.

Yes in that sense it is by design a flaky test, the buffer is to sufficient reduce the flakiness such that it would not fail on any heavily loaded system. Clearly, the buffer here is too small.

> At the very least, these tests should not attempt to verify that a wait completes "in a reasonable amount of time" when running in CI (since CI machines may be heavily loaded).

I agree these tests are not very useful in the CI. @janvorli, are you ok with removing these from the CI? Perhaps we can add them to specific new platforms as needed?

**Comment by kouvel** [comment]
When I say "these tests" I'm referring to all tests that expect a wait to complete within a certain buffered interval, there are a whole bunch of those type of tests.

**Comment by janvorli** [comment]
@kouvel 
> It's documented on Windows that waits can return before the specified duration as well.

That's true, but the doc says it is only if the `dwMilliseconds` is less than the resolution of the system clock. I actually think it meant to say that it may wait for a shorter time if the wait time is not a multiple of the clock resolution. So maybe instead of disabling the tests, we could use clock_getres to get the resolution of the clock we use for waits and only fail the test if the measured wait time was less than the expected one by that resolution (probably with some added small constant to stay on the safe side w.r.t. different clocks used for measuring the wait interval).  And we would never fail if we waited for longer than the expected time.
However, I am not sure if it is worth the hassle. We would also need to expose a new PAL function to return the resolution. 

My only concern if we disabled the test is that it is we could easily forget to enable it again during future platforms bringup.

**Comment by danmosemsft** [comment]
@kouvel maybe this can move out of 2.0 now, if you want to leave it open.
